src/main/resources/static/assets/js/scripts.js

```diff
@@ -48,32 +48,88 @@ function confirm_delete_category(id) {
             });
     });
 });
 
 function errorSaveCategory(message) {
-    document.getElementById("saveCategory").disabled = true;
-    document.getElementById("errorCategory").style.display = "block";
-    $("#newCategory #errorCategory").text(message);
+    document.getElementById("submitAddCategory").disabled = true;
+    document.getElementById("errorAddCategory").style.display = "block";
+    $("#newCategory #errorAddCategory").text(message);
+}
+
+<!-- Show a category by id in a modal form-->
+function update_category(id) {
+    axios.get('/category/' + id)
+        .then(function (response) {
+            $("#updateCategory #categoryId").val(response.data.id);
+            $("#updateCategory #categoryName").val(response.data.name);
+            $("#updateCategory #categoryDescription").val(response.data.description);
+        });
+    document.getElementById('update_category').style.display='block';
+}
+
+<!-- Validate Edit Category-->
+$(document).ready(function() {
+    $("#updateCategory #categoryName").keyup(function() {
+        document.getElementById("errorEditCategory").style.display = "none";
+        document.getElementById("submitEditCategory").disabled = false;
+        var inputName = $("#updateCategory #categoryName").val();
+        var err = validate(inputName, 3, 20, 2);
+        if (err != null) {
+            return errorEditCategory(err);
+        }
+        var id = $("#updateCategory #categoryId").val();
+        axios.get('/category/' + id)
+            .then(function (response) {
+                var name = response.data.name;
+                if (!(name.toUpperCase() === inputName.toUpperCase())) {
+                    axios.get('/categories/' + inputName)
+                        .then(function (response) {
+                            if (response.data){
+                                errorEditCategory("The name already exists! Please try something else!");
+                            };
+                        });
+                };
+            });
+    });
+});
+
+function errorEditCategory(message) {
+    document.getElementById("submitEditCategory").disabled = true;
+    document.getElementById("errorEditCategory").style.display = "block";
+    $("#updateCategory #errorEditCategory").text(message);
+}
+
+<!-- General validation function for all inputs -->
+/* level -> represent validation level
+0 - (default) validate if the word is between min and max characters
+1 - validate if the field is empty and level 0
+2 - validate level 1, level 0 and if there are empty spaces before and after
+*/
+function validate(word, min, max, level) {
+    if (word == '' && level >= 1) {
+        return "Required field! Please type something!";
+    };
+    if (word.length < min || word.length > max) {
+        return "Must be between 3 and 20 characters!";
+    };
+    if (word[0] == ' ' || word[word.length - 1] == ' ' && level >= 2) {
+        return "Please delete empty spaces before or after!";
+    };
```

Comment:
do we really need this level thing? I think it should be so much easier with a regex: /^[A-Za-z0-9_.]{3,20}$/g;